components/arm.py

```diff
@@ -0,0 +1,29 @@
+import wpilib
+
+
+class Arm:
+    """
+    Bunny grabber arm.
+    """
+
+    encoder = wpilib.AnalogPotentiometer
+    motor = wpilib.Talon
+    stop_value = 0
+    extended = False
+
+    def set_stop_value(self, stop_value):
+        self.stop_value = stop_value
+
+    def set_extended(self, extended):
```

Comment:
Can we split this into two methods, `extend()` and `retract()`? Seems a lot less awkward...

```py
arm.retract()
```
vs
```py
arm.set_extended(False)
```

Reply:
Sure. Also, what’s the best way to pass values into a component? The arms needs to know the range of encoders values to stay within, how should those be set? Currently I have a setter for them and the arm won’t move until the range is set, but that’s a little awkward.

Reply:
Can't they just be tunables? I don't see a reason that anything else should know about the positions. They just need to tell it to extend or retract.

Reply:
Also do you think we should have 2 limit switches rather than an encoder?

Reply:
I think I already asked you about that :) Yeah we should, then we won’t need to deal with the weird encoders or a pid. Also no passing around encoder limits.

Reply:
The one problem that I see is that it will be moving a lot faster when it is closer to the top position, and a lot slower near the bottom, because gravity. I don't know whether it will actually be a problem though.

Reply:
Also it will either have to go slow, or risk breaking the limit switches. I'm starting to rethink this, maybe an encoder is the best way after all.

Reply:
I'd say let's use limit switches, and if we decide it's an issue, then we can switch to an encoder.